privacy-on-beam/pbeam/count.go

```diff
@@ -68,37 +68,52 @@ type CountParams struct {
 // This aggregation is not hardened for such applications yet.
 //
 // Count transforms a PrivatePCollection<V> into a PCollection<V, int64>.
-func Count(s beam.Scope, pcol PrivatePCollection, params CountParams) beam.PCollection {
-	s = s.Scope("pbeam.Count")
+func Count(s beam.Scope, pcol PrivatePCollection, params CountParams, partitions ...beam.PCollection) beam.PCollection {
+	if len(partitions) > 1 {
+		log.Exitf("Only one partition PCollection can be specified. %v were specified.", len(partitions))
+	}
+
 	// Obtain type information from the underlying PCollection<K,V>.
 	idT, partitionT := beam.ValidateKVType(pcol.col)
+	if len(partitions) == 1 {
+		if partitionT.Type() != partitions[0].Type().Type() {
+			log.Exitf("Specified partitions must be of type %v. Got type %v instead.",
+				partitionT.Type(), partitions[0].Type().Type())
+		}
+	}
+
+	s = s.Scope("pbeam.Count")
 
 	// Get privacy parameters.
 	spec := pcol.privacySpec
 	epsilon, delta, err := spec.consumeBudget(params.Epsilon, params.Delta)
-	err = checkCountParams(params, epsilon, delta)
-	if err != nil {
-		log.Exit(err)
-	}
-
 	if err != nil {
 		log.Exitf("couldn't consume budget: %v", err)
 	}
+	err = checkCountParams(params, epsilon, delta)	
```

Comment:
Why was this moved?

Reply:
Thought it would be more important to check that partitions are of the same type before checking the parameters of count.

Reply:
I think the previous file was wrong...

Reply:
I think the previous file was wrong...